compiler/optimizations/bulkCopyRecords.cpp

```diff
@@ -135,11 +44,27 @@ static bool isAssignment(FnSymbol* fn)
 }
 
 
-static bool isTrivialAssignment(FnSymbol* fn, FnSet& visited)
+static bool isTrivialAssignment(FnSymbol* fn)
 {
   if (! isAssignment(fn))
     return false;
-  if (isTrivial(fn, visited))
+  
+  // The base argument types must match.
+  ArgSymbol* lhs = fn->getFormal(1);
+  ArgSymbol* rhs = fn->getFormal(2);
+  if (lhs->type->getValType() != rhs->type->getValType())
+    return false;
+
+  // Don't consider assignment for class types to be trivial.
+  // That is because a class type represents a pointer, so
+  // we don't really track whether or not it is POD.
+  // See
+  //   test/classes/hilde/freeAssignment.chpl
+  if (isClass(lhs->type->getValType()))
+    return false;
+
```

Comment:
I think disabling freeAssignment should be another PR if we do it. I don't think of it as a priority right now, since the optimization in question (bulkCopyRecords) wouldn't do anything useful for class types anyway. I agree that the behavior in that test isn't necessarily what we want to support.